freeipa/src/main/java/com/sequenceiq/freeipa/converter/telemetry/TelemetryConverter.java

```diff
@@ -0,0 +1,131 @@
+import com.sequenceiq.common.api.type.FeatureSetting;
+
+@Component
+public class TelemetryConverter {
+
+    private final boolean freeIpaTelemetryEnabled;
+
+    private final boolean reportDeplymentLogs;
+
+    private final String databusEndpoint;
+
+    public TelemetryConverter(@Value("${freeipa.telemetry.enabled:false}") boolean freeIpaTelemetryEnabled,
+            @Value("${cluster.deployment.logs.report:false}") boolean reportDeplymentLogs,
+            @Value("${altus.databus.endpoint:}") String databusEndpoint) {
+        this.freeIpaTelemetryEnabled = freeIpaTelemetryEnabled;
+        this.reportDeplymentLogs = reportDeplymentLogs;
+        this.databusEndpoint = databusEndpoint;
+    }
+
+    public Telemetry convert(TelemetryRequest request) {
+        Telemetry telemetry = null;
+        if (freeIpaTelemetryEnabled && request != null) {
+            telemetry = new Telemetry();
+            LoggingRequest loggingRequest = request.getLogging();
+            Logging logging = createLoggingFromRequest(loggingRequest);
+            telemetry.setLogging(logging);
+            Features features = null;
+            features = createFeaturesFromRequest(request, features);
+            telemetry.setFeatures(features);
+            telemetry.setDatabusEndpoint(databusEndpoint);
+            telemetry.setFluentAttributes(request.getFluentAttributes());
+        }
+        return telemetry;
+    }
+
+    public TelemetryResponse convert(Telemetry telemetry) {
+        TelemetryResponse response = null;
+        if (freeIpaTelemetryEnabled && telemetry != null) {
+            response = new TelemetryResponse();
+            response.setFluentAttributes(telemetry.getFluentAttributes());
+            Logging logging = telemetry.getLogging();
+            LoggingResponse loggingResponse = createLoggingResponseFromSource(logging);
+            response.setLogging(loggingResponse);
+            Features features = telemetry.getFeatures();
+            FeaturesResponse featuresResponse = createFeaturesResponseFromSource(features);
+            response.setFeatures(featuresResponse);
+        }
+        return response;
+    }
+
+    private Logging createLoggingFromRequest(LoggingRequest loggingRequest) {
+        Logging logging = null;
+        if (loggingRequest != null) {
+            logging = new Logging();
+            logging.setStorageLocation(loggingRequest.getStorageLocation());
+            if (loggingRequest.getS3() != null) {
+                S3CloudStorageV1Parameters s3Params = new S3CloudStorageV1Parameters();
+                s3Params.setInstanceProfile(loggingRequest.getS3().getInstanceProfile());
+                logging.setS3(s3Params);
+            } else if (loggingRequest.getAdlsGen2() != null) {
+                AdlsGen2CloudStorageV1Parameters adlsGen2Params = new AdlsGen2CloudStorageV1Parameters();
+                adlsGen2Params.setAccountKey(loggingRequest.getAdlsGen2().getAccountKey());
```

Comment:
Extract `loggingRequest.getAdlsGen2()` as local variable and replace all below occurrences.